d3ploy/solver.py

```diff
@@ -101,6 +101,9 @@ def preference_deploy(proto_commod, pref_fac, remainder):
             return deploy_dict
         else:
             deploy_dict[proto] += 1
+    elif remainder > 0:
+        deploy_dict[proto] = 1
+    
```

Comment:
no blank line needed. PEP8 discourages unnecessary blank lines. https://www.python.org/dev/peps/pep-000
I suggest removing this one as it doesn't add clarity.